example/simple_testing/c/diplomat_callback_structs.h

```diff
@@ -0,0 +1,17 @@
+#ifndef Diplomat_callback_structs_H
+#define Diplomat_callback_structs_H
+
+typedef struct DiplomatCallback_int32_t {
```

Comment:
if we have DiplomatCallback use an outparam for the return value we don't need to do this and we can always use `DiplomatCallback` everywhere. We've tried to move the C backend away from needing to generate such typedefs, it causes problems.

Reply:
I replied to your comment about this in the design doc, but just to duplicate it here: 

We came up with a different idea to avoid the extra file of structs: rather than generating one `DiplomatCallback` struct per return type (which is currently what we do in C), we would generate one per function argument. So, e.g., if there is a function `foo` with a callback parameter `bar` in struct `BazStruct`, then we would generate `DiplomatCallback_BazStruct_foo_bar`. 

This has a few advantages over the current approach (and which fix some of the issues you listed): 
1. we'll have the specific argument list for the `run_callback` function, instead of having a variadic function 
2. there will never be a name collision if we use multiple crates through Diplomat (which currently would happen, if we ran Diplomat on 2 crates and they both have functions with callback arguments that have the same return type)
3. no need to keep track of which callback types we have already seen, since we'd be generating one `DiplomatCallback` per callback argument 
4. no extra file with `DiplomatCallback` structs: they would be included in the same file as the other generated struct functions

The downside is that there is more duplicated code generated, but this shouldn't come at a performance cost.

Including a pointer to the output in the `run_callback` function would be doable too, but this has the downside of requiring the user to set up the output pointer in advance, and would make the use of the `DiplomatCallback` struct less like calling a normal function. Also, this would be a pain to set up with JNI for Kotlin/Java, and potentially for other non-C/C++ languages too.